### ISSUE TITLE
[core-kit] Rename unnest (probably)

### ISSUE THREAD

**Issue by aomarks** [breadboard-ai/breadboard#2616]
`unnest` is probably not the best name for this component. This component exposes all properties of the given JSON object as separate output ports (and propagates their schema from the incoming object's schema).

One thing to consider is that we should add the inverse operation too (take N input wires and create a JSON object out of them), and it would be good if it's name fit well with this one.

Options:

- unnest / nest
- split / combine
- spread / rest
- pack / unpack

**Comment by aomarks** [comment]
Side comment: I was also playing around with icons to visualize what these components do. Thinking something like this (note these don't currently work well as 48px component icons):

<img width="120" alt="image" src="https://github.com/user-attachments/assets/dfd4a0a6-f87a-48f8-b98a-f396fcc008ac">

<br>

<img width="120" alt="image" src="https://github.com/user-attachments/assets/86d38a0a-255b-41e8-b52e-a2f8dc8305da">

**Comment by dglazkov** [comment]
Yeah, that sounds right, visually. We typically use the material design icons. I found the one for joiner there. One day, we'll have an actual graphics designer. For now, I'd stick to material design's accumulated wisdom.

**Comment by aomarks** [comment]
> I found the one for joiner there.

👍 

Speaking of, `joiner` overlaps a fair bit with terminology here. Another direction is to go for slightly longer names to avoid ambiguity.

E.g. these components could be something like `splitJsonObject` and `createJsonObject`. I wonder if `joiner` could be made more precise too.
